src/modules/users/users.service.ts

```diff
@@ -35,4 +44,140 @@ export class UsersService {
       throw new Error('Failed to update user');
     }
   }
+
+  async createAdminUser(createUserDto: CreateUserDto): Promise<User> {
+    try {
+      const candidate = await this.findOne({
```

Comment:
who is "candidate"? we dont have such type of user, think about better name. code should be self-documented, naming are extremely important